elastic/kibana
# [Monitoring] Add handles graph for beats

Relates to  https://github.com/elastic/stack-monitoring/issues/3

This PR adds a graph of open handles to the beats monitoring UI. It also adds the hard and soft file handler limits to the summary bar.

<img width="1357" alt="screen shot 2018-10-19 at 10 30 52 am" src="https://user-images.githubusercontent.com/56682/47224773-58966900-d38a-11e8-8891-fce9dad1fad4.png">
<img width="709" alt="screen shot 2018-10-19 at 10 30 56 am" src="https://user-images.githubusercontent.com/56682/47224774-58966900-d38a-11e8-87af-ae4075926308.png">

## Testing steps
1) You need Filebeat running on linux (or some system that reports open handles - afaik, osx does not) so I'd recommend using a docker linux image of Filebeat: https://www.elastic.co/guide/en/beats/filebeat/current/running-on-docker.html
2) Configure Filebeat to report monitoring data by adding this to `filebeat.yml`: `xpack.monitoring.enabled: true`
3) Start Filebeat alongside ES and Kibana
4) Load the monitoring UI and navigate to the Filebeat instance view
5) See the Filebeat handles graph

## TODO

cc @ruflin @ycombinator
